File: src/main/models/SiteAuthor.ts
/**
 * 站点作者
 */

export default class SiteAuthor {
  /**
   * 主键
   */
  id: string
  /**
   * 作者来源站点id
   */
  siteId: number
  /**
   * 站点中作者的id
   */
  siteAuthorId: string
  /**
   * 站点中作者的名称
   */
  siteAuthorName: string
  /**
   * 站点中作者的曾用名
   */
  siteAuthorNameBefore: string[]
  /**
   * 介绍
   */
  introduce: string
  /**
   * 站点作者在本地对应的作者id
   */
  localAuthorId: number
  constructor(siteAuthor: SiteAuthor) {
    this.id = siteAuthor.id
    this.siteId = siteAuthor.siteId
    this.siteAuthorId = siteAuthor.siteAuthorId
    this.siteAuthorName = siteAuthor.siteAuthorName
    this.siteAuthorNameBefore = siteAuthor.siteAuthorNameBefore
    this.introduce = siteAuthor.introduce
    this.localAuthorId = siteAuthor.localAuthorId
  }
}

File: src/main-go/internal/model/site_author.go
package model

import "library-squirrel/pkg/model"

// SiteAuthor 站点作者
type SiteAuthor struct {
	model.BaseEntity
	SiteID               int64  `gorm:"column:site_id;uniqueIndex:idx_site_author_site_site_author" json:"siteId"`
	SiteAuthorID         string `gorm:"column:site_author_id;uniqueIndex:idx_site_author_site_site_author" json:"siteAuthorId"`
	AuthorName           string `gorm:"column:author_name" json:"authorName"`
	FixedAuthorName      string `gorm:"column:fixed_author_name" json:"fixedAuthorName"`
	SiteAuthorNameBefore string `gorm:"column:site_author_name_before" json:"siteAuthorNameBefore"`
	Introduce            string `gorm:"column:introduce" json:"introduce"`
	LocalAuthorID        int64  `gorm:"column:local_author_id" json:"localAuthorId"`
	LastUse              int64  `gorm:"column:last_use" json:"lastUse"`
}

func (SiteAuthor) TableName() string {
	return "site_author"
}
